libvast/include/vast/system/eraser.hpp

```diff
@@ -12,67 +12,55 @@
 
 #include "vast/ids.hpp"
 #include "vast/system/actors.hpp"
-#include "vast/system/query_processor.hpp"
+#include "vast/system/index.hpp"
+
+#include <caf/settings.hpp>
+#include <caf/typed_response_promise.hpp>
 
 #include <string>
 
 namespace vast::system {
 
 /// Periodically queries the INDEX with a configurable expression and erases
 /// all hits from the ARCHIVE.
-class eraser_state : public query_processor {
+class eraser_state {
 public:
-  // -- member types -----------------------------------------------------------
-
-  using super = system::query_processor;
-
   // -- constants --------------------------------------------------------------
 
   static inline constexpr auto name = "eraser";
 
   // -- constructors, destructors, and assignment operators --------------------
 
-  eraser_state(caf::event_based_actor* self);
-
-  void init(caf::timespan interval, std::string query, index_actor index);
+  eraser_state() = default;
 
-protected:
   // -- implementation hooks ---------------------------------------------------
 
-  void transition_to(state_name x) override;
+  [[nodiscard]] record status(status_verbosity) const;
 
-  record status(status_verbosity) override;
-
-private:
   // -- member variables -------------------------------------------------------
+  index_actor index_ = {};
 
   /// Configures the time between two query executions.
-  caf::timespan interval_;
+  caf::timespan interval_ = {};
 
   /// The query expression that selects events scheduled for deletion. Note
   /// that we get the query as string on purpose. Taking an ::expression here
   /// instead would fix any query such as `#time < 1 week ago` to the time of
   /// its parsing and not update properly.
-  std::string query_;
-
-  /// Collects hits until all deltas arrived.
-  ids hits_;
-
-  /// Keeps track whether we were triggered remotely and need to send a
-  /// confirmation message and suppress the delayed message.
-  caf::response_promise promise_;
+  std::string query_ = {};
 };
 
 /// Periodically queries `index` and erases all hits from `archive`.
 /// @param interval The time between two query executions.
 /// @param query The periodic query that selects events scheduled for deletion.
 ///              Note that we get the query as string on purpose. Taking an
-///              ::expression here instead would fix any query such as `#time <
-///              1 week ago` to the time of its parsing and not update
+///              ::expression here instead would fix any query such as
+///              `#time < 1 week ago` to the time of its parsing and not update
```

Comment:
Nit: `#time < 1 week ago` is not a valid query. Same above in the documentation of the `query_` member variable of the state.